pycds/orm/tables.py

```diff
@@ -297,6 +298,11 @@ class Variable(Base):
     derived_values = relationship("DerivedValue", back_populates="variable")
     obs_derived_values = synonym("derived_values")  # Backwards compatibility
 
+    # Constraints
+    __table_args__ = (
+            UniqueConstraint("network_id", func.lower("net_var_name"), name="network_variable_name_unique"),
```

Comment:
Here's your first SQLAlchemy gotcha: The name of the column *in the ORM* (as opposed to in the database) is `name`, not `net_var_name`. In the definition on line 282:

```
name = Column("net_var_name", String)
```

you can see the renaming. `name` is the identifier of an attribute of the ORM class; `net_var_name` is what it refers to in the database. This is not necessarily a great thing, but it's what we've got in this system.

Reply:
Oops, it's possible that I'm wrong about that. My bad. Stand by.

Reply:
> This is not necessarily a great thing, but it's what we've got in this system.

Agreed that it makes the *definition* confusing in the ORM code. The flip side is that it makes the application/usage code cleaner, as you can request the attribute as `Variable.name` rather than `Variable.net_var_name` which is nominally more wordy. (Not the *most* verbose example, but you'll find many other more egregious examples around this code base and `modelmeta`).

Reply:
@helfy18 , the actual problem is that (apparently) you can't have a function as part of a unique constraint definition. Try removing the `func.lower` and see what happens. (Run tests locally.) I'll let you do the legwork on this: Research unique constraints in PostgreSQL, and find out what valid syntax is -- can it include functions of columns. I think probably not.

Reply:
@jameshiebert , agreed.

Reply:
@jameshiebert The lower() function is not able to be inside of a unique constraint in postgres. How much does this lower() matter to the issue?

Reply:
Hmmmm, that definitely changes things. The ultimate issue here is that these names are used to create column names in a subsequent table. Column names, i.e. [*identifiers* are case in-sensitive](https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS). So ultimately we need some way of either:
+ constraining the data by some other method that respects case-insensitivity (ideas?)
+ changing [the code that does the table construction](https://github.com/pacificclimate/pycds/blob/d60277aed9b878e7287229e6822d75a905a1d4e0/pycds/alembic/versions/4a2f1879293a_create_functions.py#L216) to be case-sensitive (by using double quotes ")

The latter option is likely not a beginner task. See if with some research you can come up with something for the first option. If not, we may have to restructure this issue and bail on it for the time being.

Reply:
@jameshiebert the option that I found online is outlined here http://shuber.io/case-insensitive-unique-constraints-in-postgres/ where they suggest changing the column data type to be ```citext```, meaning case insensitive text. Not sure if this would be a possibility or not

Reply:
Cool, yeah that looks like a great option! Things to check and to include in this PR:

+ Is this extension supported by all versions of PostgreSQL that we are using?
+ Ensure the `citext` extension is added in any setup code (hint: look for anyplace where `CREATE EXTENSION POSTGIS` is used)
+ Ensure that we create a new database *migration* as part of this PR. @rod-glover may be able to help you with this.

Reply:
Can verify that citext is supported on PostgreSQL version 9.3.